pysd/py_backend/utils.py

```diff
@@ -312,38 +312,31 @@ def get_return_elements(return_columns, namespace, subscript_dict):
     --------
 
     """
-    # Todo: clean up duplicated code
     capture_elements = list()
     return_addresses = dict()
     for col in return_columns:
         if '[' in col:
             name, location = col.strip(']').split('[')
             subs = [l.strip() for l in location.split(',')]
-            try:
-                py_name = namespace[name]
-            except KeyError:
-                if name in namespace.values():
-                    py_name = name
-                else:
-                    raise KeyError(name + " not found as model element")
-
-            if py_name not in capture_elements:
-                capture_elements += [py_name]
             address = make_coord_dict(subs, subscript_dict)
-            return_addresses[col] = (py_name, address)
         else:
-            try:
-                py_name = namespace[col]
-            except KeyError:
-                if col in namespace.values():
-                    py_name = col
-                else:
-                    raise KeyError(col + " not found as model element")
-
-            if py_name not in capture_elements:
-                capture_elements += [py_name]
-            return_addresses[col] = (py_name, {})
-
+            name = col
+            address = {}
+        
+        if name in namespace:
+            py_name = namespace[name]
+        else:
+            if name in namespace.values():
+                py_name = name
+            else:
+                raise KeyError(name + " not found as model element")
+        
+        if py_name not in capture_elements:
+            capture_elements += [py_name]
+            # Don't add aliases columns to the same elements of model
+            # (f.ex. when `namespace` parameter contains multiple aliases to same element)
+            return_addresses[col] = (py_name, address)
```

Comment:
Is this to resolve the multiple Time columns issue? I would rather this function give an address for every column that is passed in the `return_columns` parameter, even the aliases, if there isn't a compelling reason otherwise. How else would you be able to resolve the alias? We should just make sure that whatever calls this function calls it asking for the correct columns. 

Does that make sense or is there something I'm missing? Its been a long time since I looked at this code... I should have done the documentation more thoroughly...

Reply:
Yep, this changes solve issue with multiple Time columns. Because in xmile and vensim parser defined few aliases to one python expression, f.ex.: 
```
time_step => time_step()
dt => time_step()
```
and others... There are I'm just remember for now. And when pysd try to return all columns it produce two columns with `time_step` and `dt` with the same values, because there reference to the same python expression, so we can use only one column for that.

Reply:
Got it. When we originally built the xmile translator, we added the `TIME STEP` alias because the integrator was expecting to use it, and we figured that if the user got both columns, it wasn't a big deal - they can always ask for the columns they want explicitly.

One problem with skipping the Aliases is that then the variable that comes first in the list will be the one that gets returned, and whatever comes after will be ignored. Its not intuitive behavior... There are cases when we might want to use aliases in the model - say for when you take a subrange of a subscript list - that's essentially an alias to the subscript set. Not sure it would interact with this code particularly, but I can't guarantee that there wouldn't be a case when getting the alias right is important.

If the only issue is having multiple columns of timestep when the user doesn't specify exactly what columns they want (which is a pretty trivial thing) then we could modify the function [`_default_return_columns`](https://github.com/JamesPHoughton/pysd/blob/master/pysd/py_backend/functions.py#L663-L674) to check for the alias, and only include one copy...


Lets flesh out your idea of having an object which keeps track of the model integration state, as then we wouldn't need to add the `TIME STEP` alias anyways...

Reply:
Hm... I think that you are right. So, OK - I'm look at this part again and try introduce another fix for that issue.

Reply:
In the meantime, if you'd like to do a PR for everything else, I'll bring it into master.

Reply:
I move changes to `_default_return_columns` function, so for now it should be more understandable for end-users.